GNUmakefile

```diff
@@ -0,0 +1,17 @@
+M := .cache/makes
+M-COMMIT ?= 683675d4c45f90f40215399ebd319805877999c9
+$(shell [ -d $M ] || ( \
+  git clone -q https://github.com/makeplus/makes $M && \
+  git -C $M reset -q --hard $(M-COMMIT)))
+ifneq ($(shell git -C $M rev-parse $(M-COMMIT)),\
+       $(shell git -C $M rev-parse HEAD))
+$(error $M is not at the correct commit: $(M-COMMIT))
+endif
+
+include $M/init.mk
+include $M/go.mk
```

Comment:
Downloading the runtime with Make is very uncommon for Go projects. The Go toolchain knows how to download the version specified in go.mod https://go.dev/doc/toolchain

Reply:
> If you want I can easily change the behavior so that it only installs a local go when GO-VERSION is provided.

Yes I think that would be best. Another option would be testing `go version` and download Go only when it's not installed.

Reply:
> Yes I think that would be best. Another option would be testing `go version` and download Go only when it's not installed.

Good idea. I'll try that out.

Reply:
Please note this should be considered instead of `go version`

`go env GOVERSION`

The result is easier to parse

Reply:
Fixed in 2acf43e